src/dataframe/dataframe.jl

```diff
@@ -502,34 +516,67 @@ end
         throw(BoundsError(df, (row_inds, col_inds)))
     end
     selected_columns = index(df)[col_inds]
-    # Computing integer indices once for all columns is faster
-    selected_rows = T === Bool ? findall(row_inds) : row_inds
-    new_columns = AbstractVector[dv[selected_rows] for dv in _columns(df)[selected_columns]]
-    return DataFrame(new_columns, Index(_names(df)[selected_columns]), copycols=false)
+
+    u = _names(df)[selected_columns]
+    lookup = Dict{Symbol, Int}(zip(u, 1:length(u)))
+    idx = Index(lookup, u)
+
+    if length(selected_columns) == 1
+        return DataFrame(AbstractVector[_columns(df)[selected_columns[1]][row_inds]], idx, copycols=false)
+    else
+        # Computing integer indices once for all columns is faster
+        selected_rows = T === Bool ? findall(row_inds) : row_inds
+        @static if VERSION >= v"1.4"
+            if length(selected_rows) >= 1_000_000 && Threads.nthreads() > 1
+                new_columns = Vector{AbstractVector}(undef, length(selected_columns))
+                @sync for i in eachindex(new_columns)
+                    Threads.@spawn new_columns[i] = _columns(df)[selected_columns[i]][selected_rows]
+                end
+                return DataFrame(new_columns, idx, copycols=false)
+            else
+                return DataFrame(AbstractVector[dv[selected_rows] for dv in view(_columns(df), selected_columns)], idx, copycols=false)
```

Comment:
This is efficient too, right?
```suggestion
                return DataFrame(AbstractVector[_columns(df)[c][selected_rows] for c in selected_columns],
                                 idx, copycols=false)
```

Reply:
Yes - this should be equivalent. I was not sure which style we would prefer. So - what do you think? Should we use double indexing or `view`?

Reply:
I find double indexing slightly simpler, but maybe that's just me.

Reply:
ok - changed